src/app/dataset-view/additional-components/metadata-component/components/set-transform/set-transform.component.html

```diff
@@ -0,0 +1,49 @@
+<div class="mb-4" *ngIf="queries">
+    <div class="text-center mt-4">
+        <p class="fs-4">Set Transformation</p>
+        <p class="text-muted">
+            Defines a transformation that produces data in a derivative dataset.
+        </p>
+    </div>
+    <div class="container mt-4">
+        <div class="row">
+            <div class="col-lg-6 left-section">
+                <app-search-section
+                    [inputDatasets]="inputDatasets"
+                    [TREE_DATA]="TREE_DATA"
+                    [dataSource]="dataSource"
+                ></app-search-section>
+                <app-engine-section
+                    [currentSetTransformEvent]="currentSetTransformEvent"
+                    (onEmitSelectedEngine)="onSelectEngine($event)"
+                ></app-engine-section>
+            </div>
+            <div class="col-lg-6 queries">
+                <app-queries-section
+                    [queries]="queries"
+                    [datasetName]="getDatasetInfoFromUrl().datasetName"
+                ></app-queries-section>
+            </div>
+        </div>
+    </div>
+    <div class="d-flex justify-content-end section-buttons">
+        <button
+            type="button"
+            mat-stroked-button
+            class="button-custom width-custom bg-primary text-white"
+            (click)="onEditYaml()"
+            data-test-id="edit-yaml-set-transform-button"
+        >
+            Edit Yaml
+        </button>
+        <button
+            type="button"
+            mat-stroked-button
+            class="button-custom ms-4 bg-success text-white width-custom"
+            (click)="onSaveEvent()"
+            data-test-id="save-set-transform-button"
+        >
+            Save
+        </button>
+    </div>
```

Comment:
Isn't this part also a component - I think we have a very similar one in SetPollingSource.

Reply:
Replace on the existing component.